### ISSUE TITLE
The name of CMi in Almagest SC does not match Toomer

### ISSUE THREAD

**Issue by abalkin** [Stellarium/stellarium#1745]
### Expected Behaviour

According to the description file, the Almagest SC is based on Toomer's translation of Almagest, so it is expected that the labels of the constellations will match those that can be found in Toomer.

### Actual Behaviour

![image](https://user-images.githubusercontent.com/535197/122079243-37040e80-cdcb-11eb-818e-bc690c3f7b39.png)

Compare this to Toomer:

```
                                [XXXIX] Constellation of Canis Minor (96)
1               The star in the neck
2               The bright star just over the hindquarters, called Procyon
                  [2 stars, 1 of the first magnitude, 1 of the fourth]
-------------------
(96) Ptolemy calls this ‘Procyon’ (Προκύων , ‘harbinger of Sirius’), after its principal star.
```

The English translation "The One before the Dog" is too long for a constellation label and is not supported by the primary source.
Since Ptolemy call CMi "after its principal star", I think it is best to use the same label "Procyon" or "Prokyon" for both the star and the constellation.

Similar issue exists with the constellation of Lyra.
 
![image](https://user-images.githubusercontent.com/535197/122082212-cf02f780-cdcd-11eb-94c4-c4a50163b241.png)

For some reason the name of the constellation is spelled differently from the name of the star" Lyre" vs. Lyra".  While "Lyre" is a proper English translation for Latin "Lyra" or Greek "λύρα", I don't see why the name of the constellation which is named after its major star should be spelled differently from the name of the star. 

### System
* Stellarium version: Stellarium 0.21.0-fa32815 [master]

**Comment by gzotti** [comment]
There is a tendency to keep star names more strictly to the Greek/Latin origin and not translate them, while a related constellation name "Lyre" may offer itself to translation. 

The question about Procyon is: what should we consider the *primary source* for the Almagest SC? The names found in the  Almagest itself, in Greek origin, where Prokyon easily translates to "the one before the dog", or Toomer's English edition, who names it "Harbinger of Sirius", the name of which is not contained in Ptolemy's wording but is given as Toomer's own explanation/interpretation. Here is Grynaeus 1538: No Sirius mentioned!

![image](https://user-images.githubusercontent.com/18099873/122090471-14441b00-ce08-11eb-8780-e25090e794b4.png)

A note in the reference what *exactly* has been taken from Toomer and where the user of this SC can expect deviations from the English standard edition, a decision of which is made by the authors of the SC (us), may be helpful. Toomer could be source for identification, or source for coordinates (when we start plotting user star lists as discussed earlier), or source for one particular sequence of stars. (Some Almagest manuscripts have copying errors.) And in later future, maybe users can switch to one of the other modern editions to compare modern scholarly editions.

**Comment by sushoff** [comment]
> so it is expected that the labels of the constellations will match those that can be found in Toomer.

yes, it is based on Toomer but it is not limited to Toomer's translation: Toomer is a scholar and his work is meant for other scholars. He does not translate the term "Procyon" but my experience with a public audience is that many people do not know to translate this term. Thus, they find it interesting to learn the literal translation. 

"Prokyon" with kappa ("k") is literally the Greek spelling - the Latin version with "c" that is used in English is certainly not Ptolemaic - of a proper name. "The One before the dog" is the literal translation.

**Comment by abalkin** [comment]
I don't have a strong view on the particular translation or transliteration, but when Toomer says that Ptolemy named a constellation after its major star, we should reflect it in the primary names.  "Before the Dog" is short enough that it can be used as a primary name for both the star and the constellation.  Similarly either "Lyra" or "Lyre" should be used for  both Lyr and Vega.

**Comment by abalkin** [comment]
>  what should we consider the primary source for the Almagest SC?

As I mentioned in the opening comment, the current description says

> To create this sky culture we used the English translation by Toomer (1984) and computer readable tables prepared by Ernie Write. 

If we decide that English variant should be based on original research into Ancient Greek sources, then this should be reflected in the description, but I don't think we have resources to perform research that will improve upon the state of the art.

I believe Stellarium  should use the best translation that is available in any given language and resort to original translations only for languages in which no published  translations exist.  At least for English, Latin, German and Russian quality translations exist and we should reuse them.

**Comment by sushoff** [comment]
As your new concept changes the concept of names, we might reflect it again. 

I considered the description "The bright star just over the hindquarters" as the usual description that I did not take as proper name. "Prokyon" is the proper name. 

Prokyon from "pro" (before) and "kyos" (dog) is the traditional name for the bright star that is found in all sources (Aratus, Eratosthenes, Hipparchus). It is not preserved whether or not authors before Ptolemy considered this asterism as a single star asterism or as a group of stars - but it is preserved that they were all native speakers in Ancient Greek! They took the term literally and understood it as "the asterism before the dog". 

Toomer is taking liberties in the translation which is fine for scholarly use but I want to make it more clear for the public. 

If you think, this way it is wrong to credit Toomer's work on which we rely we should rather change the description but not the name.

**Comment by abalkin** [comment]
> If you think, this way it is wrong to credit Toomer's work on which we rely we should rather change the description but not the name.

@sushoff - Let's focus on the names of the constellations here.  I realize that you took different approach from Toomer.  Toomer assumes that the reader is familiar with the modern (Western) names of the constellations and freely uses Latin names without translation when they are close enough to Greek original.  For example, Toomer uses modern Latin names for Ursa Minor and Major, Bootis, Corona Borealis, Draco, Cygnus, etc.  In all these examples you chose to provide your own English translations: Great and Small Bear, Ploughman, Crown, Dragon, Bird, etc.  I don't think these translations add value to for the target audience which will be exposed to Western SC before they switch to Almagest.

Yet, for the purposes of this issue, I am not arguing for a wholesale switch to Toomer name, I am just asking to correct the cases where the same name was translated differently depending on whether it is used for a star or a constellation.

Let me also mention a few other inconsistencies that probably don't deserve their own issue and can be discussed here:

1. The use of capitalization in multi-word names.  The convention seems to be to capitalize every word as in "Serpent Bearer" or "Southern Fish", but this is not done in "Sea monster",  "The ship Argo",  and ""Water snake".
2. The use of definite article ("the").  For example: "King Kepheus", but "The ship Argo".

**Comment by abalkin** [comment]
> Here is Grynaeus 1538

@gzotti - is it available somewhere on the internet?

**Comment by gzotti** [comment]
Unfortunately I cannot find it any longer. But Heiberg may be more useful. https://archive.org/details/syntaxismathema01ptolgoog
![image](https://user-images.githubusercontent.com/18099873/122128812-6c454680-ce35-11eb-90c3-ab62cae29de5.png)

**Comment by sushoff** [comment]
It should be 

- The Ship Argo (indeed, historically it is "the Ship" and "Argo" a name assigned later)
- Water Snake (typo: thanks!)
- Sea Monster (typo: thanks!) 
- but: King Kepheus (because I took the term "king" as a sort of title that is also describing)

And the modern constellation CMi should be called "Prokyon" or more consequently "[The One] Before the Dog".

**Comment by abalkin** [comment]
> BTW @abalkin : please reflect your wording when blaming other people. Instead of "I am just asking to correct the cases..." implying that the other person is wrong, you should better state "I am just suggesting to change cases ".

Sorry if I offended you – it was certainly not my intention.  I appreciate the work that you and the rest of the Stellarium team did to bring us this wonderful program and I only want to help you to make it better.  When I wrote "I am just asking to correct the cases...", I meant to say "I am just asking to *allow me to* correct the cases..." implying that I stand ready to submit a pull request once we reach a consensus.

**Comment by abalkin** [comment]
>  it is not all wrong that you don't understand.

Astronomy is my hobby rather than profession and I readily agree that there are things in Archeoastronomy that I don't understand.  On the other hand, you wrote above that "Toomer is a scholar and his work is meant for other scholars." implying that Stellarium is targeting wider audience than professors and graduate students in Archeoastronomy.   Where I as a non-scholar see an inconsistency, it is likely that most Stellarium users will see it as well even if there is a historically sound argument justifying that inconsistency.

> historically it is "the Ship" and "Argo" a name assigned later

Is this an argument for keeping the article "The" in the name?  Why is "Hero Orion" written without "The", but "The Ship Argo" with?  Is it because the name of the constellation is simply "Argo" and "the ship Argo"   is a phrase explaining that "Argo" was a ship? 

> And the modern constellation CMi should be called "Prokyon" or more consequently "[The One/Constellation/Asterism] Before the Dog".

I think we should strike a balance between scholarly precision and aesthetic appeal.  When it comes to primary labels, brevity and consistency is more important than completeness.  My key argument is that if Hipparchus uses the same name for the star and the constellation, so should Stellarium.

Currently, the CMi/CMa area is displayed as follows:

![image](https://user-images.githubusercontent.com/535197/122259094-c67af180-ce9f-11eb-990e-70933325a39a.png)

I would prefer seeing

```
              Prokyon
                    * Προκύων (Prokyon)
```

Note that adding "Constellation/Asterism]" is redundant because the constellation and star labels are already distinguished by color.

If we cannot assume that "Prokyon" is close enough to English "Procyon" that users will recognize it without translation, then we can display

```
              Prokyon (Before the Dog)
                    * Prokyon (Before the Dog)
```

This would be a more substantial change and would require changing primary labels for  a few other stars, but in general I think Greek variants should not be displayed together with user language translations in Greek script rather than transliteration.  It would be better to add "Ancient Greek" as a choice in the sky culture language menu and move  Ancient Greek variants there.

**Comment by gzotti** [comment]
> in general I think Greek variants should not be displayed together with user language translations in Greek script rather than transliteration

Sorry, but this discussion simply comes a few years early. We have already voiced plans to enhance the skyculture topic with more options towards forming labels from selectable elements: original glyphs, user language dependent transliteration, translation, and optionally showing modern names. Can we please all accept that this skyculture is currently not solvable into perfection when two or three persons follow diverging objectives. There is one version included now which is authored by the team. It may not be perfect, but we need to add critical infrastructure first. For the time being, we can decide how to mix original lettering and translations. 

But Indeed, we should follow the wording of the original author, Ptolemy, as close as seems best in times of limited screen space. 

"Argo" is just written like that in the Syntaxis (in Heiberg and Grynaeus), no "Ship" included. (Actually, all constellations are written like "asterism of ...".) If earlier authors named it "ship" without "Argo", this may be annotated in a different skyculture, but is then not relevant here in the Almagest skyculture. The proper mapping of "Argous asterismos" [simple transliteration, sorry. Think Greek here.] should still be "Argo" for brevity (to avoid screen clutter), or we must write "Constellation of Argo", "Constellation of the Lyre", ... everywhere. This would on the other hand help discern "Constellation of the Lyre" from "The bright one on the turtle shell, named Lyra". 

It is nothing to decide during the next few days and before the upcoming release, but should be kept in mind for future revisions.

**Comment by abalkin** [comment]
>  Can we please all accept that this skyculture is currently not solvable into perfection when two or three persons follow diverging objectives. 

I am not sure there are diverging objectives, and if they do exists, I have not seen them clearly articulated.  My objective with respect to Almagest SC is to make Stellarium suitable for teaching about history of Astronomy in general and Almagest specifically.  I don't see where this objective can diverge with that of original authors.  It is possible that when it comes to teaching we have different audiences in mind.  In my case the audience is English/Russian speaking middle schoolers.  For @sushoff, the audience may be older and more sophisticated, but I don't why my desire to make English labels closer the way the appear in English literature is in conflict with the expectations of a more sophisticated user.

As I mentioned, my primary languages are English and Russian and I think Russian labels should not be literal translations of English labels, but taken from a reputable edition of Almagest in Russian.  (Fortunately, a very good translation has been published in 1998.)

Here is an example of what happens when a name of an asterism  is translated from Greek to English and then to Russian.  "Plokamos", for which Toomer uses the Latin name "Coma" is currently translated as "Lock" in English which in turn translated as "Замок" to Russian.   "Замок" in Russian means "lock" as in "lock and key" and is very different from the Ptolemy's original.

To summarize, my goal is to make Stellarium displays of Almagest SC to be faithful visualization of the modern English and Russian editions of Almagest.  If this conflicts with the objectives of others, it will be helpful to vocalize those objectives and hopefully we will be able to find a solution that will work for all.

**Comment by gzotti** [comment]
re "Lock": This is the problem with context free and unguided translations, which is even worse in the context of skycultures. Transifex translators may not be familiar with the context of particular words. I face the same problem as German reviewer. Also, I am pretty sure large skycultures like the Chinese already have "canonical" translations into various languages in 20th-century textbooks which however most Transifexers don't know.  If you find such mistranslations, please join Transifex and correct them. 

@sushoff Can you please change the one sentence in the introduction:

>To create this sky culture we used the English translation by Toomer (1984). 

into something along

>To create this sky culture we used the English translation by Toomer (1984) but changed a few formulations where we saw need.

so that abalkin does not wrongly believe this tries to be a 1:1 representation of Toomer.

@abalkin if you need a 1:1 representation of Toomer now, you may need to create your own SC for your teaching. To repeat myself, currently some of the technical infrastructure that we would like to have is not here yet. Annotating names with sources, and selection of allowed sources, will be able to switch names as we need them. But in fact, I am still on a different building site.

**Comment by abalkin** [comment]
> If you find such mistranslations, please join Transifex and correct them.

I already did this for the "The Lock [in Leo]" string.  For translating the many recent additions of Almagest descriptions to Russian, I find  Transifex interface very inconvenient.  Can I contribute a .po file instead?  I am currently trying to figure out Transifex API, so if you have any pointers for the bulk uploads – let me know.

**Comment by abalkin** [comment]
> @sushoff Can you please change the one sentence in the introduction:
> 
> > To create this sky culture we used the English translation by Toomer (1984).
> 
> into something along
> 
> > To create this sky culture we used the English translation by Toomer (1984) but changed a few formulations where we saw need.

While we are at it, could you please also update the Toomer reference to 1998 edition. According to Erinie Write, "The transcriptions were reconciled with the second edition of Toomer (Princeton University Press, 1998), which is also the source of the star descriptions."

**Comment by abalkin** [comment]
> @abalkin if you need a 1:1 representation of Toomer now, you may need to create your own SC for your teaching. 

I was considering doing that while #1647 was pending, but with that PR merged, all I really need for my teaching is to complete Russian translation.  Hopefully my reuse of [Veselovsky](https://ru.wikipedia.org/wiki/%D0%92%D0%B5%D1%81%D0%B5%D0%BB%D0%BE%D0%B2%D1%81%D0%BA%D0%B8%D0%B9,_%D0%98%D0%B2%D0%B0%D0%BD_%D0%9D%D0%B8%D0%BA%D0%BE%D0%BB%D0%B0%D0%B5%D0%B2%D0%B8%D1%87) for that will not face as much resistance as my reuse of Toomer.

**Comment by gzotti** [comment]
It's preferrable to refer to the edition which was actually used. Updating to a new edition is yet another task.

**Comment by abalkin** [comment]
> It's preferrable to refer to the edition which was actually used

 #1647 is certainly based on the 1998 edition because is was generated from  Erinie Write's files.  Since the original authors provided very few of the 1,024 star descriptions I doubt that any of these descriptions changed between the editions of Toomer and if they did,  #1647 changed them to match Erinie Write's files and the 1998 edition.

**Comment by alex-w** [comment]
> As I mentioned, my primary languages are English and Russian and I think Russian labels should not be literal translations of English labels, but taken from a reputable edition of Almagest in Russian. (Fortunately, a very good translation has been published in 1998.)

I've found this translation! Thanks for suggestion @abalkin 

> For translating the many recent additions of Almagest descriptions to Russian, I find Transifex interface very inconvenient. 

Did you use filters at Transifex WUI?

> Can I contribute a .po file instead? 

Transifex allow upload .po files, but probably not for generic translators. Please check resources section.

**Comment by sushoff** [comment]
> sushoff Can you please change the one sentence in the introduction:

done

**Comment by gzotti** [comment]
FYI: Grynaeus:  http://dx.doi.org/10.3931/e-rara-13800
